File: boj/10866.cpp
#include <iostream>
#include <deque>

using namespace std;

bool isEmpty(deque<int> dq) {
    if (dq.empty()) {
        cout << -1 << "\n";
        return true;
    } else {
        return false;
    }
}

void solve() {
    deque<int> dq;
    int n;

    cin >> n;
    for (int i = 0; i < n; i++) {
        string str;

        cin >> str;
        if (str == "push_back") {
            int num;

            cin >> num;
            dq.push_back(num);
        } else if (str == "push_front") {
            int num;

            cin >> num;
            dq.push_front(num);
        } else if (str == "front") {
            if (!isEmpty(dq))
                cout << dq.front() << "\n";
        } else if (str == "back") {
            if (!isEmpty(dq))            
                cout << dq.back() << "\n";
        } else if (str == "size") {
            cout << dq.size() << "\n";
        } else if (str == "empty") {
            if (dq.empty()) {
                cout << 1 << "\n";
            } else {
                cout << 0 << "\n";
            }
        } else if (str == "pop_front") {
            if (!isEmpty(dq)) {
                cout << dq.front() << "\n";
                dq.pop_front();
            }
        } else if (str == "pop_back") {
            if (!isEmpty(dq)) {
                cout << dq.back() << "\n";                
                dq.pop_back();
            }
        }
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    solve();
}

File: boj/10866.py
from collections import deque
import sys

input = sys.stdin.readline
N = int(input())
dq = deque()
for _ in range(N):
    cmd = input().split()
    if cmd[0] == "push_back":
        dq.append(cmd[1])
    elif cmd[0] == "push_front":
        dq.appendleft(cmd[1])
    elif cmd[0] == "pop_front":
        print(dq.popleft()) if dq else print(-1)
    elif cmd[0] == "pop_back":
        print(dq.pop()) if dq else print(-1)
    elif cmd[0] == "size":
        print(len(dq))
    elif cmd[0] == "empty":
        print(0) if dq else print(1)
    elif cmd[0] == "front":
        print(dq[0]) if dq else print(-1)
    elif cmd[0] == "back":
        print(dq[-1]) if dq else print(-1)
